tracer/src/Datadog.Trace/Agent/DiscoveryService/DiscoveryService.cs

```diff
@@ -254,6 +262,13 @@ int GetNextSleepDuration(int? previousDuration) =>
 
         private async Task ProcessDiscoveryResponse(IApiResponse response)
         {
+            // Extract and store container tags hash from response headers
+            var containerTagsHash = response.GetHeader(AgentHttpHeaderNames.ContainerTagsHash);
+            if (containerTagsHash != null)
+            {
+                ContainerMetadata.ContainerTagsHash = containerTagsHash;
```

Comment:
I swear you're trolling me with all this static stuff 😅 Having static readonly state is one thing, but having static _mutable_ state is a whole new ball game 😛  I think the `ContainerMetadata` object likely needs refactoring to be an instance type and passed in to where it's used, so that we don't destroy our tests completely. I'd suggest doing that in a separate PR first, then rebasing this on top

Reply:
ok fair 😅